File: packages/js-sdk/src/regex/detectors/ip-address.ts
// IP address detector (IPv4 and IPv6)

import { Detector } from '../base'
import { EntityType, PIIMatch } from '../entities'
import { registerUniversal } from '../registry'

const IPV4 = /\b(?:(?:25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}(?:25[0-5]|2[0-4]\d|[01]?\d\d?)\b/g

const IPV6 = new RegExp(
  '\\b(?:[0-9a-fA-F]{1,4}:){7}[0-9a-fA-F]{1,4}\\b'
  + '|\\b(?:[0-9a-fA-F]{1,4}:){1,7}:\\b'
  + '|\\b::(?:[0-9a-fA-F]{1,4}:){0,5}[0-9a-fA-F]{1,4}\\b',
  'g'
)

class IpAddressDetector extends Detector {
  entityType = EntityType.IP_ADDRESS
  score = 0.95

  iterMatches(text: string): PIIMatch[] {
    const results: PIIMatch[] = []

    for (const pattern of [IPV4, IPV6]) {
      pattern.lastIndex = 0
      let match: RegExpExecArray | null
      while ((match = pattern.exec(text)) !== null) {
        const matchedText = match[0]
        const start = match.index
        // Skip common non-IP patterns like version numbers
        if (pattern === IPV4 && IpAddressDetector.isVersionNumber(text, start)) {
          continue
        }
        results.push({
          entityType: this.entityType,
          text: matchedText,
          start,
          end: start + matchedText.length,
          score: this.score,
        })
      }
    }

    return results
  }

  /** Skip patterns preceded by 'v' or 'version' that look like version numbers. */
  private static isVersionNumber(text: string, start: number): boolean {
    if (start > 0 && text[start - 1].toLowerCase() === 'v') {
      return true
    }
    const prefix = text.slice(Math.max(0, start - 10), start).toLowerCase().trim()
    return prefix.endsWith('version')
  }
}

registerUniversal(IpAddressDetector)

export { IpAddressDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/ip_address.py
"""IP address detector (IPv4 and IPv6)"""

import re
from typing import List

from ..base import Detector, RegexDetector
from ..entities import EntityType, PIIMatch
from ..registry import register_universal

_IPV4 = re.compile(
    r"\b(?:(?:25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}(?:25[0-5]|2[0-4]\d|[01]?\d\d?)\b"
)

_IPV6 = re.compile(
    r"\b(?:[0-9a-fA-F]{1,4}:){7}[0-9a-fA-F]{1,4}\b"
    r"|\b(?:[0-9a-fA-F]{1,4}:){1,7}:\b"
    r"|\b::(?:[0-9a-fA-F]{1,4}:){0,5}[0-9a-fA-F]{1,4}\b"
)


@register_universal
class IpAddressDetector(Detector):
    entity_type = EntityType.IP_ADDRESS
    score = 0.95

    def iter_matches(self, text: str) -> List[PIIMatch]:
        results: List[PIIMatch] = []
        for pattern in (_IPV4, _IPV6):
            for match in pattern.finditer(text):
                matched_text = match.group()
                # Skip common non-IP patterns like version numbers
                if pattern is _IPV4 and self._is_version_number(text, match.start()):
                    continue
                results.append(PIIMatch(
                    entity_type=self.entity_type.value,
                    text=matched_text,
                    start=match.start(),
                    end=match.end(),
                    score=self.score,
                ))
        return results

    @staticmethod
    def _is_version_number(text: str, start: int) -> bool:
        """Skip patterns preceded by 'v' or 'version' that look like version numbers."""
        if start > 0 and text[start - 1].lower() == "v":
            return True
        prefix = text[max(0, start - 10):start].lower().strip()
        return prefix.endswith("version")
